pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java

```diff
@@ -92,6 +96,36 @@ protected ZooKeeperCache localZkCache() {
         return pulsar().getLocalZkCache();
     }
 
+    // bookkeeper zookeeper
+    protected ZooKeeper bookieZk() {
+        ZooKeeper zkClient = null;
+
+        ServiceConfiguration config = pulsar().getConfiguration();
+
+        String bookkeeperMetadataServiceUri = config.getBookkeeperMetadataServiceUri();
+        URI uri = URI.create(bookkeeperMetadataServiceUri);
+        String bookieZkConnect = StringUtils.replace(uri.getAuthority(), ";", ",") + uri.getPath();
+
+        int zkTimeout = (int) config.getZooKeeperSessionTimeoutMillis();
+        try {
+            zkClient = ZooKeeperClient.newBuilder().connectString(bookieZkConnect)
```

Comment:
umm.. we are removing zk direct dependency from the pulsar broker so, not a good idea to introduce new one. we should use `MetadataStore` instead.

Reply:
yeah, but it depends on the refactoring of the "ZkIsolatedBookieEnsemblePlacementPolicy" and "ZkBookieRackAffinityMapping" first.